### PR TITLE
External dependencies upgrade - Spring Boot 3.4.0 and Spring Cloud 2024.0.0

### PR DESCRIPTION
**Pull request by Netyyyy**
Updates external dependencies to align with Spring Boot version [3.4.0](https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-dependencies/3.4.0/spring-boot-dependencies-3.4.0.pom) from [3.3.5](https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-dependencies/3.3.5/spring-boot-dependencies-3.3.5.pom) and Spring Cloud version [2024.0.0](https://repo1.maven.org/maven2/org/springframework/cloud/spring-cloud-dependencies/2024.0.0/spring-cloud-dependencies-2024.0.0.pom) from [2023.0.3](https://repo1.maven.org/maven2/org/springframework/cloud/spring-cloud-dependencies/2023.0.3/spring-cloud-dependencies-2023.0.3.pom).
<details><summary>Release notes</summary><p><em>Sourced from <a href='https://github.com/spring-projects/spring-boot/releases/tag/v3.4.0'>spring-boot releases</a>.</em></p><h4>⭐ New Features</h4><ul> <li>Add withDefaultRequestConfigCustomizer method to HttpComponentsClientHttpRequestFactoryBuilder <a href="https://github.com/spring-projects/spring-boot/issues/43139" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43139/hovercard">#43139</a></li> <li>Fail JsonWriter if duplicate names are detected <a href="https://github.com/spring-projects/spring-boot/issues/43041" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43041/hovercard">#43041</a></li> <li>Add JsonObjectDeserializer.nullSafeValue method that accepts a mapper Function <a href="https://github.com/spring-projects/spring-boot/issues/42972" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/42972/hovercard">#42972</a></li> <li>Support timeout property for GraphQL over SSE <a href="https://github.com/spring-projects/spring-boot/pull/42966" data-hovercard-type="pull_request" data-hovercard-url="/spring-projects/spring-boot/pull/42966/hovercard">#42966</a></li> <li>Improve performance of ConfigurationPropertiesBinder by storing bind handlers on first access <a href="https://github.com/spring-projects/spring-boot/issues/42950" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/42950/hovercard">#42950</a></li> <li>Improve performance of ConcurrentReferenceCachingMetadataReaderFactory <a href="https://github.com/spring-projects/spring-boot/issues/42949" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/42949/hovercard">#42949</a></li> <li>Log warning in HikariCheckpointRestoreLifecycle if pool suspension isn't configured <a href="https://github.com/spring-projects/spring-boot/pull/42937" data-hovercard-type="pull_request" data-hovercard-url="/spring-projects/spring-boot/pull/42937/hovercard">#42937</a></li> <li>Remove spring-boot-starter-aop dependency from spring-boot-starter-data-jpa and spring-boot-starter-integration <a href="https://github.com/spring-projects/spring-boot/issues/42934" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/42934/hovercard">#42934</a></li></ul><h4>🐞 Bug Fixes</h4><ul> <li>Jersey body handling is inconsistent with Spring Webflux and Spring MVC <a href="https://github.com/spring-projects/spring-boot/issues/43209" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43209/hovercard">#43209</a></li> <li>Classes are accidentally named "structure logging" instead of "structured logging" <a href="https://github.com/spring-projects/spring-boot/pull/43203" data-hovercard-type="pull_request" data-hovercard-url="/spring-projects/spring-boot/pull/43203/hovercard">#43203</a></li> <li>StructuredLoggingJsonProperties customizer should be a Class reference rather than a String <a href="https://github.com/spring-projects/spring-boot/issues/43202" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43202/hovercard">#43202</a></li> <li>Cannot package OCI image when 'docker.io/paketobuildpacks/new-relic' is provided as a buildpack <a href="https://github.com/spring-projects/spring-boot/issues/43171" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43171/hovercard">#43171</a></li> <li>Incorrect Type for 'management.endpoints.access.default' defined in additional-spring-configuration-metadata.json <a href="https://github.com/spring-projects/spring-boot/issues/43154" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43154/hovercard">#43154</a></li> <li>WebServerPortFileWriter fails when using a portfile without extension <a href="https://github.com/spring-projects/spring-boot/issues/43117" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43117/hovercard">#43117</a></li> <li>SslOptions.isSpecified() only returns true if ciphers and enabled protocols are set <a href="https://github.com/spring-projects/spring-boot/issues/43084" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43084/hovercard">#43084</a></li> <li>SslHealthIndicator throws NullPointerException when using SslBundle with SslStoreBundle.NONE <a href="https://github.com/spring-projects/spring-boot/issues/43078" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43078/hovercard">#43078</a></li> <li>JdkClientHttpRequestFactoryBuilder and JettyClientHttpRequestFactoryBuilder do not set Ciphers or Enabled Protocols <a href="https://github.com/spring-projects/spring-boot/issues/43077" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43077/hovercard">#43077</a></li> <li>Root cause of errors is hidden when loading images from archive <a href="https://github.com/spring-projects/spring-boot/issues/43070" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43070/hovercard">#43070</a></li> <li>mvn spring-boot:run fails on Windows with "Could Not Find or Load Main Class" when path contains non-ASCII characters <a href="https://github.com/spring-projects/spring-boot/issues/43062" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43062/hovercard">#43062</a></li> <li>A <code>@SpyBean</code> on the output of a FactoryBean is not reset <a href="https://github.com/spring-projects/spring-boot/issues/43053" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43053/hovercard">#43053</a></li> <li>Logback logging system does not process URLs with paths not ending in .xml <a href="https://github.com/spring-projects/spring-boot/issues/42990" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/42990/hovercard">#42990</a></li> <li>Bean-based conditions do not consider factory beans correctly when determining if they are a candidate <a href="https://github.com/spring-projects/spring-boot/issues/42970" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/42970/hovercard">#42970</a></li> <li>NPE in bootBuildImage when setting DOCKER_CONTEXT=default <a href="https://github.com/spring-projects/spring-boot/issues/42960" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/42960/hovercard">#42960</a></li> <li>Warning due to duplicate MockResolver extensions <a href="https://github.com/spring-projects/spring-boot/issues/42957" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/42957/hovercard">#42957</a></li> <li>HttpHostConnectException is thrown when using buildpacks with Gradle or Maven on Windows <a href="https://github.com/spring-projects/spring-boot/issues/42952" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/42952/hovercard">#42952</a></li> <li>build-info doesn't support seconds since the epoch from project.build.outputTimestamp <a href="https://github.com/spring-projects/spring-boot/issues/42936" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/42936/hovercard">#42936</a></li> <li>NPE in OnClassCondition.resolveOutcomesThreaded following thread interruption because firstHalf is null <a href="https://github.com/spring-projects/spring-boot/issues/42926" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/42926/hovercard">#42926</a></li> <li>Default WebSocketMessageBrokerConfigurer is always overriding custom channel executor <a href="https://github.com/spring-projects/spring-boot/issues/42924" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/42924/hovercard">#42924</a></li> <li>X-Registry-Auth header sent to Docker Engine API contains field "authHeader" <a href="https://github.com/spring-projects/spring-boot/issues/42915" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/42915/hovercard">#42915</a></li> <li>ApplicationContextRunner has inconsistent behaviour with duplicate auto-configuration class names <a href="https://github.com/spring-projects/spring-boot/issues/17963" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/17963/hovercard">#17963</a></li></ul><h4>📔 Documentation</h4><ul> <li>Migrate class references to full javadoc links <a href="https://github.com/spring-projects/spring-boot/issues/43239" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43239/hovercard">#43239</a></li> <li>Documentation for 'spring.datasource.type' is misleading <a href="https://github.com/spring-projects/spring-boot/issues/43199" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43199/hovercard">#43199</a></li> <li>Update "Upgrading From" section to use "2.x" <a href="https://github.com/spring-projects/spring-boot/issues/43160" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43160/hovercard">#43160</a></li> <li>Include spring-boot-loader in API documentation <a href="https://github.com/spring-projects/spring-boot/issues/43153" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43153/hovercard">#43153</a></li> <li>Document how and where to add custom GraalVM configuration files <a href="https://github.com/spring-projects/spring-boot/issues/43074" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43074/hovercard">#43074</a></li> <li>Rework DataSource configuration examples to separate defining an additional DataSource and defining a DataSource of a different type <a href="https://github.com/spring-projects/spring-boot/issues/43059" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43059/hovercard">#43059</a></li> <li>Location of the layers schema is incorrect in the Maven Plugin's examples <a href="https://github.com/spring-projects/spring-boot/issues/43033" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43033/hovercard">#43033</a></li> <li>Link to Eclipse setup instructions <a href="https://github.com/spring-projects/spring-boot/issues/42954" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/42954/hovercard">#42954</a></li> <li>Fix link to Checkpoint and Restore status page <a href="https://github.com/spring-projects/spring-boot/issues/42939" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/42939/hovercard">#42939</a></li></ul><h4>🔨 Dependency Upgrades</h4><ul> <li>Upgrade to ActiveMQ 6.1.4 <a href="https://github.com/spring-projects/spring-boot/issues/43128" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43128/hovercard">#43128</a></li> <li>Upgrade to Byte Buddy 1.15.10 <a href="https://github.com/spring-projects/spring-boot/issues/43097" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43097/hovercard">#43097</a></li> <li>Upgrade to Couchbase Client 3.7.5 <a href="https://github.com/spring-projects/spring-boot/issues/43098" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43098/hovercard">#43098</a></li> <li>Upgrade to Elasticsearch Client 8.15.4 <a href="https://github.com/spring-projects/spring-boot/issues/43129" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43129/hovercard">#43129</a></li> <li>Upgrade to Flyway 10.20.1 <a href="https://github.com/spring-projects/spring-boot/issues/43130" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43130/hovercard">#43130</a></li> <li>Upgrade to Groovy 4.0.24 <a href="https://github.com/spring-projects/spring-boot/issues/43099" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43099/hovercard">#43099</a></li> <li>Upgrade to Hibernate 6.6.2.Final <a href="https://github.com/spring-projects/spring-boot/issues/43100" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43100/hovercard">#43100</a></li> <li>Upgrade to HttpClient5 5.4.1 <a href="https://github.com/spring-projects/spring-boot/issues/43102" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43102/hovercard">#43102</a></li> <li>Upgrade to Infinispan 15.0.11.Final <a href="https://github.com/spring-projects/spring-boot/issues/43131" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43131/hovercard">#43131</a></li> <li>Upgrade to Jackson Bom 2.18.1 <a href="https://github.com/spring-projects/spring-boot/issues/43103" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43103/hovercard">#43103</a></li> <li>Upgrade to Jetty 12.0.15 <a href="https://github.com/spring-projects/spring-boot/issues/43104" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43104/hovercard">#43104</a></li> <li>Upgrade to jOOQ 3.19.15 <a href="https://github.com/spring-projects/spring-boot/issues/43105" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43105/hovercard">#43105</a></li> <li>Upgrade to Kafka 3.8.1 <a href="https://github.com/spring-projects/spring-boot/issues/43106" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43106/hovercard">#43106</a></li> <li>Upgrade to Lettuce 6.4.1.RELEASE <a href="https://github.com/spring-projects/spring-boot/issues/43185" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43185/hovercard">#43185</a></li> <li>Upgrade to Logback 1.5.12 <a href="https://github.com/spring-projects/spring-boot/issues/43107" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43107/hovercard">#43107</a></li> <li>Upgrade to Lombok 1.18.36 <a href="https://github.com/spring-projects/spring-boot/issues/43186" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43186/hovercard">#43186</a></li> <li>Upgrade to Maven Dependency Plugin 3.8.1 <a href="https://github.com/spring-projects/spring-boot/issues/43108" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43108/hovercard">#43108</a></li> <li>Upgrade to Maven Failsafe Plugin 3.5.2 <a href="https://github.com/spring-projects/spring-boot/issues/43109" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43109/hovercard">#43109</a></li> <li>Upgrade to Maven Surefire Plugin 3.5.2 <a href="https://github.com/spring-projects/spring-boot/issues/43110" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43110/hovercard">#43110</a></li> <li>Upgrade to Micrometer 1.14.1 <a href="https://github.com/spring-projects/spring-boot/issues/43187" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43187/hovercard">#43187</a></li> <li>Upgrade to Micrometer Tracing 1.4.0 <a href="https://github.com/spring-projects/spring-boot/issues/43120" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43120/hovercard">#43120</a></li> <li>Upgrade to MongoDB 5.2.1 <a href="https://github.com/spring-projects/spring-boot/issues/43111" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43111/hovercard">#43111</a></li> <li>Upgrade to Netty 4.1.115.Final <a href="https://github.com/spring-projects/spring-boot/issues/43133" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43133/hovercard">#43133</a></li> <li>Upgrade to Prometheus Client 1.3.3 <a href="https://github.com/spring-projects/spring-boot/issues/43112" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43112/hovercard">#43112</a></li> <li>Upgrade to Pulsar Reactive 0.5.9 <a href="https://github.com/spring-projects/spring-boot/issues/43188" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43188/hovercard">#43188</a></li> <li>Upgrade to Reactor Bom 2024.0.0 <a href="https://github.com/spring-projects/spring-boot/issues/43015" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43015/hovercard">#43015</a></li> <li>Upgrade to Spring AMQP 3.2.0 <a href="https://github.com/spring-projects/spring-boot/issues/43016" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43016/hovercard">#43016</a></li> <li>Upgrade to Spring Authorization Server 1.4.0 <a href="https://github.com/spring-projects/spring-boot/issues/43017" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43017/hovercard">#43017</a></li> <li>Upgrade to Spring Batch 5.2.0 <a href="https://github.com/spring-projects/spring-boot/issues/43018" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43018/hovercard">#43018</a></li> <li>Upgrade to Spring Data Bom 2024.1.0 <a href="https://github.com/spring-projects/spring-boot/issues/43019" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43019/hovercard">#43019</a></li> <li>Upgrade to Spring Framework 6.2.0 <a href="https://github.com/spring-projects/spring-boot/issues/43020" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43020/hovercard">#43020</a></li> <li>Upgrade to Spring HATEOAS 2.4.0 <a href="https://github.com/spring-projects/spring-boot/issues/43021" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43021/hovercard">#43021</a></li> <li>Upgrade to Spring Integration 6.4.0 <a href="https://github.com/spring-projects/spring-boot/issues/43022" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43022/hovercard">#43022</a></li> <li>Upgrade to Spring Kafka 3.3.0 <a href="https://github.com/spring-projects/spring-boot/issues/43023" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43023/hovercard">#43023</a></li> <li>Upgrade to Spring LDAP 3.2.8 <a href="https://github.com/spring-projects/spring-boot/issues/43189" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43189/hovercard">#43189</a></li> <li>Upgrade to Spring Pulsar 1.2.0 <a href="https://github.com/spring-projects/spring-boot/issues/43024" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43024/hovercard">#43024</a></li> <li>Upgrade to Spring RESTDocs 3.0.3 <a href="https://github.com/spring-projects/spring-boot/issues/43025" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43025/hovercard">#43025</a></li> <li>Upgrade to Spring Security 6.4.1 <a href="https://github.com/spring-projects/spring-boot/issues/43232" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43232/hovercard">#43232</a></li> <li>Upgrade to Spring Session 3.4.0 <a href="https://github.com/spring-projects/spring-boot/issues/43027" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43027/hovercard">#43027</a></li> <li>Upgrade to Testcontainers 1.20.4 <a href="https://github.com/spring-projects/spring-boot/issues/43243" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43243/hovercard">#43243</a></li> <li>Upgrade to Tomcat 10.1.33 <a href="https://github.com/spring-projects/spring-boot/issues/43134" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43134/hovercard">#43134</a></li> <li>Upgrade to Undertow 2.3.18.Final <a href="https://github.com/spring-projects/spring-boot/issues/43166" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43166/hovercard">#43166</a></li> <li>Upgrade to WebJars Locator Lite 1.0.1 <a href="https://github.com/spring-projects/spring-boot/issues/43135" data-hovercard-type="issue" data-hovercard-url="/spring-projects/spring-boot/issues/43135/hovercard">#43135</a></li></ul><h4>❤️ Contributors</h4><p>Thank you to all the contributors who worked on this release:</p><p><a class="user-mention notranslate" data-hovercard-type="user" data-hovercard-url="/users/ahoehma/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/ahoehma">@ahoehma</a>, <a class="user-mention notranslate" data-hovercard-type="user" data-hovercard-url="/users/deki/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/deki">@deki</a>, <a class="user-mention notranslate" data-hovercard-type="user" data-hovercard-url="/users/izeye/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/izeye">@izeye</a>, <a class="user-mention notranslate" data-hovercard-type="user" data-hovercard-url="/users/ngocnhan-tran1996/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/ngocnhan-tran1996">@ngocnhan-tran1996</a>, <a class="user-mention notranslate" data-hovercard-type="user" data-hovercard-url="/users/nosan/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/nosan">@nosan</a>, <a class="user-mention notranslate" data-hovercard-type="user" data-hovercard-url="/users/quaff/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/quaff">@quaff</a>, and <a class="user-mention notranslate" data-hovercard-type="user" data-hovercard-url="/users/wickdynex/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/wickdynex">@wickdynex</a></p></details>

This PR is created by GitHub Actions: https://github.com/Azure/spring-cloud-azure-tools/actions/runs/12150436995

### CODE REVIEW

**Review by saragluna — APPROVED** [review]

**Comment by Netyyyy** [comment]
@alzimmermsft @jeanbisutti please help take a review, thanks

**Review by alzimmermsft — APPROVED** [review]

File: sdk/spring/spring-cloud-azure-docker-compose/pom.xml [hunk 1]
```diff
@@ -63,6 +63,12 @@
       <groupId>com.azure</groupId>
       <artifactId>azure-storage-blob</artifactId>
       <version>12.29.0</version> <!-- {x-version-update;com.azure:azure-storage-blob;dependency} -->
+      <exclusions>
+        <exclusion>
+          <groupId>com.fasterxml.jackson.core</groupId>
+          <artifactId>jackson-core</artifactId>
+        </exclusion>
+      </exclusions>
```
Comment (srnagar) [inline sdk/spring/spring-cloud-azure-docker-compose/pom.xml:71]: Why do we need to exclude `jackson-core` here?

Comment (Netyyyy) [reply sdk/spring/spring-cloud-azure-docker-compose/pom.xml:71]: It introduces conflicting dependencies, which can cause `NoSuchMethodError: ParserMinimalBase` error

**Review by srnagar — APPROVED** [review]

File: sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/CosmosRepositoryFactory.java [hunk 2]
```diff
@@ -54,11 +54,13 @@ public <T, ID> EntityInformation<T, ID> getEntityInformation(Class<T> domainType
     }
 
     @Override
+    @SuppressWarnings("deprecation")
```
Comment (rujche) [inline sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/CosmosRepositoryFactory.java:57]: Do you have plan to remove the usage of deprecated things?

**Review by rujche — APPROVED** [review]

**Review by jeanbisutti — APPROVED** [review]

File: sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/CosmosRepositoryFactory.java (hunk 2 above)
Comment (Netyyyy) [reply sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/CosmosRepositoryFactory.java:57]: maybe new major version or it was removed, no definite plan

**Pull request merged**